tests/ctest/external-messaging2-toit-run-test.c

```diff
@@ -43,19 +50,36 @@ static toit_err_t on_rpc_request(void* user_data, int sender, int function, toit
 }
 
 static toit_err_t on_removed(void* user_data) {
-  printf("freeing user context\n");
+  test_service_t* test_service = (test_service_t*)user_data;
+  printf("freeing user data %d\n", test_service->id);
   free(user_data);
   return TOIT_ERR_SUCCESS;
 }
 
 static void __attribute__((constructor)) init() {
-  printf("registering external handler\n");
-  test_handler_t* test_handler = (test_handler_t*)malloc(sizeof(test_handler_t));
+  printf("registering external handler 0\n");
+  test_service_t* test_service = (test_service_t*)malloc(sizeof(test_service_t));
+  test_service->id = 0;
+  test_service->msg_context = NULL;
   toit_msg_cbs_t cbs = {
     .on_created = &on_created,
     .on_message = &on_message,
     .on_rpc_request = &on_rpc_request,
     .on_removed = &on_removed,
   };
-  toit_msg_add_handler("toit.io/external-test", test_handler, cbs);
+  toit_msg_add_handler("toit.io/external-test0", test_service, cbs);
+}
+
+static void __attribute__((constructor)) init2() {
+  printf("registering external handler 1\n");
+  test_service_t* test_service = (test_service_t*)malloc(sizeof(test_service_t));
```

Comment:
Having a `toit_malloc` that does GC if necessary would make this a bit more robust. It actually doesn't matter here, because this only runs on startup, but it is something to consider for the other places where we need to allocate.